test/image_builder/codebase/test_processes.py

```diff
@@ -0,0 +1,62 @@
+from pathlib import Path
+
+import pytest
+from pyfakefs.fake_filesystem_unittest import TestCase
+
+from image_builder.codebase.processes import CodebaseProcessNoProcfileError
+from image_builder.codebase.processes import load_codebase_processes
+
+
+class TestCodebaseProcesses(TestCase):
+    def setUp(self):
+        self.setUpPyfakefs()
+
+    def test_without_procfile_present(self):
+        with pytest.raises(CodebaseProcessNoProcfileError):
+            load_codebase_processes(Path("."))
+
+    def test_loading_simple_processes(self):
+        self.fs.create_file(
+            "Procfile",
+            contents="web: cd src && python manage.py migrate --noinput && waitress-serve "
+            "--port=$PORT --threads=6 config.wsgi:application",
+        )
+
+        processes = load_codebase_processes(Path("."))
+
+        assert processes[0].name == "web"
+        assert "cd src" in processes[0].commands
+        assert "python manage.py migrate --noinput" in processes[0].commands
+        assert (
+            "waitress-serve --port=$PORT --threads=6 config.wsgi:application"
+            in processes[0].commands
+        )
```

Comment:
Wondering if we care about the order of these commands?

E.g.
```suggestion
        self.assertEqual(processes[0].name, "web")
        self.assertEqual(processes[0].commands[0], "cd src")
        self.assertEqual(processes[0].commands[1], "python manage.py migrate --noinput")
        self.assertEqual(
            processes[0].commands[2],
            "waitress-serve --port=$PORT --threads=6 config.wsgi:application",
        )
```

Reply:
Fixed this instance and a few others.